bazel/ray_deps_setup.bzl

```diff
@@ -1,80 +1,80 @@
 load("@bazel_tools//tools/build_defs/repo:git.bzl", "git_repository", "new_git_repository")
 load("@bazel_tools//tools/build_defs/repo:http.bzl", "http_archive", "http_file")
 
-def github_repository(*, name=None, remote=None, commit=None, tag=None,
-                      branch=None, build_file=None, build_file_content=None,
-                      sha256=None, archive_suffix=".zip", shallow_since=None,
-                      strip_prefix=True, url=None, path=None, **kwargs):
+def urlsplit(url):
+    """ Splits a URL like "https://example.com/a/b?c=d&e#f" into a tuple:
+        ("https", ["example", "com"], ["a", "b"], ["c=d", "e"], "f")
```

Comment:
can we use dict? https://docs.bazel.build/versions/2.0.0/skylark/lib/dict.html

Reply:
I can if you want, but I thought it's nicer to keep them in order and somewhat similar to Python's [own `urlsplit` (which returns a tuple)](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlsplit).

Reply:
yeah but python returns named tuple which are a bit more readable when retrieving the result.

Reply:
Haha yeah true, ok sure I'll change it.